contracts/src/Agent.sol

```diff
@@ -14,9 +16,13 @@ contract Agent {
     /// @dev The gateway contract controlling this agent
     address public immutable GATEWAY;
 
-    constructor(bytes32 agentID) {
+    /// @dev The para_id of the agent
+    ParaID public immutable PARA_ID;
```

Comment:
Not all agents are toplevel parachains. Agents can also correspond to nested consensus systems within a parachain, such as pallets or ink! smart contracts.

For this reason, it does not make much sense for an agent to have a paraid field.

Reply:
Yes, It's not required for this PR.  But when start with some design for arbitrary transact from Ethereum -> Parachain, 
a new external function in https://linear.app/snowfork/issue/SNO-615 in `Gateway` we need to find the matching ParaId for the agent.

Reply:
Yes, it's not required for this PR, but seems nessessary for arbitrary transact from Ethereum to Polkadot some initial design in https://linear.app/snowfork/issue/SNO-615
For `sendTransact(bytes32 agentId, bytes memory payload, uint256 extraFee)` , we need to find the matching ParaId for the particular agent.

Reply:
Cool, though I added some advice on that in https://linear.app/snowfork/issue/SNO-615#comment-48a711d1

Reply:
Reverted the change about paraid